javers-core/src/main/java/org/javers/core/JaversBuilder.java

```diff
@@ -91,29 +95,51 @@ public Javers build() {
      * @see <a href="http://javers.org/documentation/configuration/#repository-setup">http://javers.org/documentation/configuration/#repository-setup</a>
      */
     public JaversBuilder registerJaversRepository(JaversRepository repository){
-        Validate.argumentsAreNotNull(repository);
+        argumentsAreNotNull(repository);
         this.repository = repository;
         return this;
     }
 
     /**
-     * registers an {@link Entity} with id-property pointed by @Id annotation
+     * Registers an {@link Entity}.
+     * Use @Id annotation to mark exactly one Id-property.
+     * <br/><br/>
+     *
+     * Optionally, use @Transient or @{@link DiffIgnore} to mark ignored properties.
      *
      * @see <a href="http://javers.org/documentation/configuration/#domain-model-mapping">http://javers.org/documentation/configuration/#domain-model-mapping</a>
      */
     public JaversBuilder registerEntity(Class<?> entityClass) {
-        Validate.argumentIsNotNull(entityClass);
+        argumentIsNotNull(entityClass);
         return registerEntity( new EntityDefinition(entityClass));
     }
 
     /**
-     * registers an {@link Entity} with id-property selected explicitly by name
+     * Registers an {@link Entity}. <br/>
+     * Use this method if you are not willing to use annotations.
      *
+     * @param idPropertyName mandatory, name of Id-property
+     * @param ignoredProperties optional, list of property names to be ignore by JaVers
      * @see <a href="http://javers.org/documentation/configuration/#domain-model-mapping">http://javers.org/documentation/configuration/#domain-model-mapping</a>
      */
-    public JaversBuilder registerEntity(Class<?> entityClass, String idPropertyName) {
-        Validate.argumentsAreNotNull(entityClass, idPropertyName);
-        return registerEntity( new EntityDefinition(entityClass, idPropertyName) );
+    public JaversBuilder registerEntity(Class<?> entityClass, String idPropertyName, List<String> ignoredProperties) {
+        argumentsAreNotNull(entityClass, idPropertyName);
+
+        List<String> ignored = Collections.EMPTY_LIST;
+        if (ignoredProperties != null){
+            ignored = ignoredProperties;
+        }
+        return registerEntity( new EntityDefinition(entityClass, idPropertyName, ignored) );
+    }
+
+    /**
+     * Deprecated since 1.0.6,
+     * use {@link #registerEntity(Class)}  or
+     * {@link #registerEntity(Class, String, java.util.List)}
+     */
+    @Deprecated
```

Comment:
dlaczego ```@deprecated``` ? jeżeli ktoś będzie chciał zarejestrować entity bez ignorowanych to ma to zrobić w ten sposób:

```
registerEntity(DummyUser.class, "id", Collections.EMPTY_LIST)
``` 

? zdecydowanie wygodniej byłoby nie podawać tej pustej listy na końcu, moim zdaniem deprecated nie potrzebne